cmd/charm/charmcmd/listresources.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the GPLv3, see LICENCE file for details.
+
+package charmcmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/juju/cmd"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"launchpad.net/gnuflag"
+)
+
+var info = cmd.Info{
+	Name:    "list-resources",
+	Args:    "<charm> [--channel <channel>]",
+	Purpose: "display the resources for a charm in the charm store",
+	Doc: `
+This command will report the resources for a charm in the charm store.
+
+<charm> can be a charm URL, or an unambiguously condensed form of it,
+just like the deploy command. So the following forms will be accepted:
+
+For cs:trusty/mysql
+  mysql
+  trusty/mysql
+
+For cs:~user/trusty/mysql
+  cs:~user/mysql
+
+Where the series is not supplied, the series from your local host is used.
+Thus the above examples imply that the local series is trusty.
+`,
+}
+
+type ListResourcesCharmstoreClient interface {
+	ListResources([]*charm.URL) (map[string][]params.Resource, error)
+	SaveJAR() error
+}
+
+type NewCharmstoreClientFn func(*cmd.Context, string, string) (ListResourcesCharmstoreClient, error)
```

Comment:
Can you put names on the arguments?  Otherwise I have no idea what the two strings are for.